site/content/protocols/data-disclosure-agreement/1.0/readme.md

```diff
@@ -0,0 +1,718 @@
+### Pre-requisite
+
+The prerequisites are:
+
+1. All the actors involved in the protocol flow are DIDComm agents. Each interaction between the actors requires a DIDComm message.
+   
+2. Prior to the PS-SDA protocol flow, actors must establish connections [1] between each other and identify themselves by presenting the necessary proofs [2].
+   
+3. All the actors will have a wallet (mobile or cloud-based, individual and potentially for an enterprise) address and an associated public / private key pair.
+   
+4. A wallet address is an externally owned account (EOA) address. Ethereum addresses are composed of the prefix "0x" (hexadecimal) concatenated with the rightmost 20 bytes of the Keccak-256 hash of the ECDSA public key with the curve secp256k1.
+
+### Out of scope
+
+The administrative and governance frameworks of the “marketplace” or “data space” is out of scope. The marketplace has performed the proof of business and onboarded DS and DUS. 
+
+### Publish DDA offer to Marketplace (DS)
+
+![](../images/publish-dda-to-marketplace.png)
+#### Messages
+
+The `/dda-marketplace/1.0` protocol consists of these messages:
+
+1. `/dda-marketplace/1.0/publish-request`
+2. `/dda-marketplace/1.0/publish-response`
+
+#### Sign the DDA offer
+
+The DS signs the DDA offer using the secp256k1 private key. The signature is prepared by performing the proof algorithm described by W3C Data Integrity 1.0 specification [3].
+#### Send signed DDA offer to Data Intermediary
+
+The DS construct a DIDComm message of type `/dda-marketplace/1.0/publish-request` and sends it to the Data Intermediary DIDComm agent. An example is provided below.
+
+```json
+{
+  "@type": "https://didcomm.org/dda/1.0/publish-request",
+  "@id": "999f6c2b-b0e5-4123-aab0-b5f7bfc780c4",
+  "created_time": "1639288911",
+  "@from": "<sender did>",
+  "@to": "<receipient did>",
+  "body": {
+    "data_disclosure_agreement": {}
+  }
+}
+```
+#### Data Intermediary stores the signed DDA offer to CAS
+
+Once the DDA is received from the DS, the Data Intermediary stores the offer in Content Addressable Storage (CAS) for e.g. IPFS and obtains the Content Identifier (CID). 
+#### Data Intermediary anchors CID to Ethereum
+
+The Content Identifier (CID) pointing to a specific DDA offer is anchored to Ethereum by executing the `publishDDACID` smart contract function.
+
+The Data Intermediary create a /dda-marketplace/1.0/publish-response DIDComm message and sends it to the DS. This message contains Ethereum transaction hash for the DS to verify the transaction. An example is provided below:
+
+```json
+{
+  "@type": "https://didcomm.org/dda/1.0/publish-response",
+  "@id": "53f19e0b-5be2-480a-92bc-fcdeabf69ad3",
+  "~thread": {
+        "thid": "999f6c2b-b0e5-4123-aab0-b5f7bfc780c4"
+    },
+  "created_time": "1639288911",
+  "@from": "<sender did>",
+  "@to": "<receipient did>",
+  "body": {
+    "txn_hash": "0xeba2df809e7a612a0a0d444ccfa5c839624bdc00dd29e3340d46df3870f8a30e"
+  }
+}
```

Comment:
I think you are using envelops from DIDComm V1. They should be adapted to [DIDComm v2 format](https://identity.foundation/didcomm-messaging/spec/v2.0/#plaintext-message-structure)

Reply:
Updated to DIDComm V2 envelope.